src/config_brain_memory.js

```diff
@@ -188,7 +188,7 @@ Upgrade less: ${strings.upgradeLess}
 };
 
 brain.prepareMemory = function() {
-  Memory.username = Memory.username || _.find(Game.spawns, 'owner').owner.username;
+  Memory.username = Memory.username || _.chain(Game.rooms).map('controller').flatten().filter('my').map('owner.username').first().value();
```

Comment:
hum, shoundn't crash as it is save in memory, and didn't get error destroying it. but if you want, this should be lighter :
`_(Game.rooms).find(room => room.controller.my).value().controller.owner.username`